src/irmin/irmin.mli

```diff
@@ -158,7 +159,9 @@ module Backend : sig
   (** The complete collection of backend implementations. *)
 end
 
-module Key = Key
+module Storage = Storage
+(** [Storage] provides {!Storage.Make} for defining custom storage layers that
```

Comment:
Just pushed some changes to this. Going to revisit tomorrow after a night to sleep on it.

Part of the complication that I see is that `Backend` is used in two ways in `irmin.mli`. One is that to name a container of things that relate to backends (of which `Storage` could arguably be placed into) and two it is the specific signature, `Backend.S`, for use with `Of_backend`. I don't really want to put `Storage` into the "container" `Backend` since then it would sit next to `Backend.S` which I think might be more confusing.